apps/ensindexer/src/plugins/subregistry/shared/lib/get-current-registration.ts

```diff
@@ -0,0 +1,30 @@
+import type { Context } from "ponder:registry";
+import schema from "ponder:schema";
+
+import type { Node } from "@ensnode/ensnode-sdk";
+
+import {
+  buildSubregistryRegistration,
+  type SubregistryRegistration,
+} from "@/lib/subregistry/registration";
+
+/**
+ * Get currently indexed subregistry registration for a node.
+ */
+export async function getCurrentRegistration(
+  context: Context,
+  node: Node,
+): Promise<SubregistryRegistration> {
+  const currentRegistration = await context.db.find(schema.subregistry_registration, {
+    node,
+  });
+
+  // Invariant: the current registration must exists for the renewal to take place.
+  if (!currentRegistration) {
+    throw new Error(
+      `The current registration for "${node}" node must exist for the renewal to take place.`,
+    );
+  }
```

Comment:
```suggestion
  // Invariant: the current registration must exist.
  if (!currentRegistration) {
    throw new Error(
      `The current registration for "${node}" node does not exist.`,
    );
  }
```

Goal: don't tie the logic in this utility function exclusively to renewals.